### DISCUSSION TITLE
Does strongswan support ENCRYPTION_ALGORITHM AES_CTR (key size 256)

### DISCUSSION THREAD

**Discussion by leelamohan-p** [strongswan/strongswan#1106] [General]
**System (please complete the following information):**
 - strongSwan version(s): 5.7.1

**Describe the bug**
IPSec tunnel is not established when network selected proposal was ENCRYPTION_ALGORITHM AES_CTR (key size 256) 

**To Reproduce**
1. Enable CTR_256 Encryption algorithm in UE/Mobile. Network has to select CTR_256 Encryption algorithm

**Expected behavior**
IPSec Tunnel has to be established

**Logs/Backtraces**
06-10 10:55:38.263  1435  1681 D IPSECD  : IKE algo[aes128-aes192-aes256-aes128ctr-aes192ctr-aes256ctr-null-aesxcbc-md5-sha1-sha2_256-modp2048-], str length[91]
06-10 10:55:38.867 12560 12570 I charon  : 07[KEY] [select_proposal()  372] selected proposal: IKE:AES_CTR_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_512/MODP_2048
06-10 10:55:38.910 12560 12570 I charon  : 07[IKE] [derive_ike_traditional()  221] ENCRYPTION_ALGORITHM AES_CTR (key size 256) not supported!
06-10 10:55:38.910 12560 12570 I charon  : 07[IKE] [process_i() 1046] key derivation failed

**Additional context**

**Comment by tobiasbrunner** [comment]
To use the algorithm for IKE, you need a plugin that provides it, see [here](https://docs.strongswan.org/docs/5.9/config/IKEv2CipherSuites.html#_encryption_algorithms) for details.
